src/traces/scatter3d/attributes.js

```diff
@@ -155,11 +156,13 @@ var attrs = module.exports = overrideAll({
     ),
 
     textposition: extendFlat({}, scatterAttrs.textposition, {dflt: 'top center'}),
-    textfont: {
-        color: scatterAttrs.textfont.color,
-        size: scatterAttrs.textfont.size,
-        family: extendFlat({}, scatterAttrs.textfont.family, {arrayOk: false})
-    },
+    textfont: fontAttrs({
+        editType: 'calc',
+        colorEditType: 'style',
+        arrayOk: true,
+        variantValues: ['normal', 'small-caps'],
```

Comment:
Is there a reason for limiting the variants here?

Or, should we maybe just only support these two variants everywhere?

Not sure just wondering.

Reply:
Aha, this is because of WebGL I'm guessing?

Maybe make it a global constant value like `webGLSupportedVariants`?

Reply:
WebGL traces different mechanism to render text i.e. via CanvasRenderingContext2D https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font

At the moment only some of those options are supported.
So we should have it this way so that for those specific traces and subplots (gl2d, gl3d and mapbox) the `variant` options are limited.

Reply:
> Aha, this is because of WebGL I'm guessing?
> 
> Maybe make it a global constant value like `webGLSupportedVariants`?

Yes because of WebGL pipelines. 
But no to have a global variable since `scattergl`, `gl2d`, `gl3d` and `mapbox` each uses different pipeline. 
At one point we may be able to add one option to one not the other.
So I think it's better to keep it this way.